homework/list.js

```diff
@@ -1,44 +1,51 @@
 'use strict';
 
-(function() {
+(function(exports) {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+var ListManager = function() {
+  this._listNoteContent = [];
+  this._wrapper =document.querySelector('#note-list-wrapper');
+};
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+ListManager.prototype = {
+
+  start() {
+      this.fetchList().then((function(response) {
+      this.updateList(response);
+      this.drawList();
+      this.preloadFirstNote();
+    }).bind(this)).catch((function() {  
+	}).bind(this));
+    
+
+    window.addEventListener('click', (function(event) {
+      this.onNoteOpen(event);
+    }).bind(this));
+  },
 
-  function onNoteOpen(event) {
+  onNoteOpen(event) {
     if (event.target.classList.contains('note-title')) {
       var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
+      var content = this._listNoteContent[id];
       window.dispatchEvent(new CustomEvent('note-open',
         { detail: content }));
     };
-  }
+  },
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
+  preloadFirstNote() {
+    if (this._listNoteContent.length !== 0) {
```

Comment:
nit: It's better to put the condition outside a function wholly wrapped by only one branch.